### PR TITLE
Gobierto Data Scaffolding

### PR DESCRIPTION
**Pull request by jorgeatgu**
Closes #N

## :v: What does this PR do?

Scaffolding

For this module, I've created a structure similar to Nuxt. Now, we have three folders, one of them for pages like Datasets, another folder for Layout, and the last one for components. [link](https://github.com/PopulateTools/gobierto/tree/2743-gobierto-data-front/app/javascript/gobierto_data/webapp)

**Pages:** inside here, I've included every tab for sidebar(queries, sets, and categories) and any section of the top menu. 

**Layout:** at this moment, I've created one Layout. In this Layout, we have one section for the sidebar and the other for main content.

**Components:** We've folders for every tab and another folder for commons components like resources, queries, etc...

I a few doubts about the naming of the components; We've names like _queries_ that can be [tab](https://github.com/PopulateTools/gobierto/blob/2743-gobierto-data-front/app/javascript/gobierto_data/webapp/pages/Queries.vue), another [tab](https://github.com/PopulateTools/gobierto/blob/2743-gobierto-data-front/app/javascript/gobierto_data/webapp/components/sets/Queries.vue), and [component](https://github.com/PopulateTools/gobierto/blob/2743-gobierto-data-front/app/javascript/gobierto_data/webapp/components/commons/Queries.vue) and option menu.

## :mag: How should this be manually tested?
Nope

### CODE REVIEW

**Comment by Crashillo** [comment]
First off, fix the conflicts, then resolve all the Hound complaints (if done, click on *resolve conversation*, to set as solved)

Hound-bot reads the configuration files pointed [here](https://github.com/PopulateTools/gobierto/blob/master/.hound.yml). It's a good practice you install such plugins in your IDE, to let you know any problem before pushing. Also, you may run [this npm scripts](https://github.com/PopulateTools/gobierto/blob/4d805b66a289005ce85b7fc023300ac87ac4bc92/package.json#L70).

**Comment by jorgeatgu** [comment]
> Hound

Done!

File: app/javascript/gobierto_data/webapp/pages/Home.vue [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        />
+        <DataSets />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import Nav from "./../components/Nav.vue";
+import DataSets from "./../pages/DataSets.vue";
```
Comment (ferblape) [inline app/javascript/gobierto_data/webapp/pages/Home.vue]: Module not found: Error: [CaseSensitivePathsPlugin] `gobierto/app/javascript/gobierto_data/webapp/pages/DataSets.vue` does not match the corresponding path on disk `Datasets.vue`.

File: app/assets/stylesheets/module-data.scss [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+
+  &-summary-header {
+    padding: 1rem;
+    box-sizing: border-box;
+  }
+
+  &-summary-header-container {
+    margin: .5rem 0;
+
+    .fas {
```
Comment (Crashillo) [inline app/assets/stylesheets/module-data.scss]: May you avoid the use of this vendor class?

File: app/assets/stylesheets/module-data.scss [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    }
+  }
+
+  &-summary-queries-panel-title {
+    font-weight: bold;
+    font-size: 16px;
+    line-height: 20px;
+    color: #666;
+    display: inline-block;
+
```
Comment (Crashillo) [inline app/assets/stylesheets/module-data.scss]: Remove empty lines

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      <div class="gobierto-data-summary-header-container">
+        <i class="fas fa-tag" />
+        <span class="gobierto-data-summary-header-container-label">
+          {{ labelSubject }}
+        </span>
+        <a
+          href=""
+          class="gobierto-data-summary-header-container-text-link"
+        >
+          Urbanismo e infraestructuras
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Info.vue]: Literal string. Shouldn't be a translated text?

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+<script>
+export default {
+  name: "Info",
+  data() {
+    return {
+      labelUpdated: '',
+      labelFrequency: '',
+      labelSubject: '',
+      labelDownloadData: '',
+      datasetDescription: 'Este conjuntos de datos contiene el detalle de más de 1.200 elementos para actividades de mayores de la ciudad de Madrid con su tipología y coordenadas. En este portal tambien están disponibles otros.'
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Info.vue:53]: Literal string. Shouldn't be a translated text?

File: app/javascript/gobierto_data/webapp/components/commons/Resources.vue [hunk 6]
```diff
@@ -0,0 +1,31 @@
+<template>
+  <div class="gobierto-data-summary-resources">
+    <h2 class="gobierto-data-tabs-section-title">
+      <i class="fas fa-caret-down" />
+      Recursos
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/commons/Resources.vue]: Literal string. Shouldn't be a translated text?

File: app/javascript/gobierto_data/webapp/components/sets/Nav.vue [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+          :class="{ 'is-active': activeTab === 4 }"
+          class="gobierto-data-sets-nav--tab"
+          @click="activateTab(4)"
+        >
+          <span>{{ labelDownload }}</span>
+        </li>
+      </ul>
+    </nav>
+    <Summary v-if="activeTab === 0" />
+    <Data v-if="activeTab === 1" />
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/sets/Nav.vue]: If the activeTab isn't never gonna be multiple, use `v-else-if` better

File: app/javascript/gobierto_data/webapp/components/sets/Summary.vue [hunk 8]
```diff
@@ -0,0 +1,141 @@
+<template>
+  <div>
+    <Info />
+    <div class="gobierto-data-container-btn-download-data">
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/components/sets/Summary.vue]: Convert this button into a component (DownloadButton) if you're gonna use several times.

File: app/javascript/gobierto_data/webapp/pages/Categories.vue [hunk 9]
```diff
@@ -0,0 +1,30 @@
+<template>
+  <div class="pure-u-1 pure-u-lg-3-4 gobierto-data-layout-column">
+    <div style="background-color: rgba(113, 184, 193, .1); color: #71B8C1; height: 40vh; widht: 100%; padding: 1rem; font-weight: bold;">
```
Comment (Crashillo) [inline app/javascript/gobierto_data/webapp/pages/Categories.vue:3]: Are these styles temporal? Mark as solved if so.

File: config/environments/development.rb [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
 Rails.application.configure do
   # Verifies that versions and hashed value of the package contents in the project's package.json
-  config.webpacker.check_yarn_integrity = true
+  config.webpacker.check_yarn_integrity = false
```
Comment (Crashillo) [inline config/environments/development.rb]: @ferblape should we turn this off?

**Review by Crashillo — CHANGES_REQUESTED** [review]

**Comment by Crashillo** [comment]
After this PR, I would appreciate smaller PRs 😇

File: app/javascript/gobierto_data/webapp/components/commons/Info.vue (hunk 5 above)
Comment (jorgeatgu) [reply app/javascript/gobierto_data/webapp/components/commons/Info.vue:53]: It's a fake and temporal string.

**Comment by jorgeatgu** [comment]
Updated PR. Focused on SQL Editor's workflow. @blat @Crashillo 

[SQL Editor Header](https://github.com/PopulateTools/gobierto/blob/2743-gobierto-data-front/app/javascript/gobierto_data/webapp/components/sets/SQLEditorHeader.vue)

[SQL Editor CodeMirror ](https://github.com/PopulateTools/gobierto/blob/2743-gobierto-data-front/app/javascript/gobierto_data/webapp/components/sets/SQLEditorCode.vue)

**Comment by ferblape** [comment]
@jorgeatgu I still see the naming issue with datasets.vue component:

Screen Shot 2020-01-17 at 14 59 30

Also, on Tuesday we'll show you how to check the CircleCI to detect issues: https://app.circleci.com/jobs/github/PopulateTools/gobierto/16015

**Pull request merged**